### PR TITLE
Support for FBX embedded textures

### PR DESCRIPTION
**Pull request by OtgerRP**
Reads the embedded textures in FBX files which are referenced by the materials, and adds them to aiScene->mTextures (fixes issue #48).

The texture paths in the aiMaterial properties are set following the example of the COLLADA importer as `"*0"`, `"*1"`, etc. (this is documented anywhere?)

It works on the models I tested (embedded .png/.tga), but might be missing some cases, and in particular support for the LayeredTexture.

Also added a property to disable loading them (AI_CONFIG_IMPORT_FBX_READ_TEXTURES).

### CODE REVIEW

**Comment by OtgerRP** [comment]
Seems that some tests fail because the binary FBX models had textures embedded, and now they appear at the dumps.

File: code/FBXDocument.h [hunk 1]
```diff
@@ -654,6 +662,59 @@ typedef std::fbx_unordered_map<std::string, const Texture*> TextureMap;
 typedef std::fbx_unordered_map<std::string, const LayeredTexture*> LayeredTextureMap;
 
 
+/** DOM class for generic FBX videos */
+class Video : public Object
+{
+public:
+
+    Video(uint64_t id, const Element& element, const Document& doc, const std::string& name);
+    ~Video();
```
Comment (kimkulling) [inline code/FBXDocument.h]: Not virtual.

**Comment by kimkulling** [comment]
Seems so, can you update the dump in your pull request, please?

File: code/FBXDocument.h (hunk 1 above)
Comment (OtgerRP) [inline code/FBXDocument.h]: I just did copy the Texture class above it, which neither has it, and wanted to keep it similar. In any case they are meant to be "final" classes, so virtual destructor is not essential.

**Comment by OtgerRP** [comment]
Done!

With your permission I also added a few doc lines to aiTexture explaining how the embedded textures are referenced from the materials (did not find anything about it).

**Comment by OtgerRP** [comment]
I also added a quick fix to #652. FBX parser returned a parse error when it found a "PP" type connection, now it just ignores them.

**Comment by kimkulling** [comment]
Merged, thanks a lot!

Kimmi

**Pull request merged**